### ISSUE TITLE
[fix] Close modal button cannot be overridden

### ISSUE THREAD

**Issue by jaredcunha** [trussworks/react-uswds#3092]
**ReactUSWDS Version & USWDS Version:**

**Describe the bug**

The text to close the modal cannot be overridden, so in a spanish translated site, the control is announced in English on screen readers.

**Expected behavior**

A prop can be provided to override close button name. This would also be helpful for overriding the english copy as well. 

**Screenshots**

**Additional context**

**Device and Browser Information (please complete the following information if describing a UI bug):**
